### DISCUSSION TITLE
Fahrzeugidentifikation?

### DISCUSSION THREAD

**Discussion by kirchner85** [goecharger/go-eCharger-API-v2#182] [General]
Hallo zusammen,
ich bin auf der Suche, bzw. in der Hoffnung, dass ich irgendwie über die API herausfinden kann, welches spezifische Fahrzeug angeschlossen ist. Die Idee dahinter ist, eine Beschränkung umzusetzten, welche nicht über RFID gelöst werden muss. Meine Vorstellung ist, dass ich eine eindeutige Fahrzeug-ID erhalte und diese mit einer Liste abgleichen kann. Bisher konnte ich solch eine ID jedoch nicht über die API finden. Da es schon etwas länger her ist, weiß ich gerade nicht mehr, welche API-keys ich bereits ausprobiert habe. Blöderweise habe ich auch aktuell nur ein EV, daher kann ich keine Gegenprüfung durchführen.

Über Standort-Daten des Fahrzeugs habe ich es versucht, diese werden aber blöderweise zu selten und sehr unzuverlässig aktualisiert (Renault Twingo). Da die Einfahrt auch recht breit ist, könnte sich bei Abwesenheit trotzdem jemand am anderen Fahrzeug vorbei zwängen.

Hat da jemand eine Lösung, oder auch gerne eine andere Idee? Zur Automatisierung nutzte ich NodeRed, sofern von Relevanz.
Viele Grüße.

**Comment by 0xFEEDC0DE64** [comment]
Die go-eCharger haben kein ISO15118 und damit keine digitale Kommunikation zum Fahrzeug, es wird lediglich mit einer PWM Pulsebreite die erlaubte Stromstärke kommuniziert, und das Auto kann durch belasten dieses Signals kommunizieren, dass es gerne die Spannung durchgeschalten hätte.

Hätten wir ISO15118 im Charger, dann hätten wir viele Möglichkeiten das Auto zu identifizieren, unter anderem die MAC Adresse oder IPv6 des Fahrzeugs, irgendwelche eindeutigen Identifier im Protokoll, oder einfach das ISO15118 Feature "Plug&Charge".

**Reply by SolarPower2024** [reply]
> Die go-eCharger haben kein ISO15118 und damit keine digitale Kommunikation zum Fahrzeug, es wird lediglich mit einer PWM Pulsebreite die erlaubte Stromstärke kommuniziert, und das Auto kann durch belasten dieses Signals kommunizieren, dass es gerne die Spannung durchgeschalten hätte.
> 
> Hätten wir ISO15118 im Charger, dann hätten wir viele Möglichkeiten das Auto zu identifizieren, unter anderem die MAC Adresse oder IPv6 des Fahrzeugs, irgendwelche eindeutigen Identifier im Protokoll, oder einfach das ISO15118 Feature "Plug&Charge".

Wie ich deiner Antwort entnehmen kann, findet keine digitale Kommunikation statt. Was verständlich ist, ohne ISO15118.. 
Aber für was ist dann der neue key "plugandchargeidentifer" in der neuen Firmware? Der ist dann ja nutzlos oder?

![Screenshot_20250712_092623_go-e~2](https://github.com/user-attachments/assets/91cd9d83-bb06-4d7a-a926-d4af64ac624f)

**Reply by schinagl** [reply]
Vl ist das für den neuen go-e pro der Ende q3 ausgeliefert werden soll. 
Der kann theoretisch viel mehr als der jetzige go-e

Jedenfalls hoff ich es denn ich hab einen bestellt ☺️

**Comment by mleck28** [comment]
Hallo kirchner85,

bist du noch an einer Fahrzeugerkennung interessiert?

**Reply by YoBnEb** [reply]
Hi @mleck28,
falls du eine Möglichkeit gefunden hast, dann wäre ich da auch ganz stark daran interessiert.

Danke schon mal!

**Reply by Joemoe134** [reply]
Ich wäre auch an einer Lösung interessiert. Wenn ich die Identität des angeschlossenen Fahrzeugs für eine Entsperrungslogik einsetzen könnte, könnte ich mir die Einbindung der KIA Connect app und das Auslesen der Fahrzeuginformationen via Kia verzichten.
Ich habe eine go-e Charger Gemini flex 2.0

**Comment by kirchner85** [comment]
Hi,zum aktuellen Zeitpunkt benötige ich es nicht mehr zwingend, aber würde es, sofern sowas zur Verfügung stehen würde, trotzdem (irgendwie) nutzen.Viele Grüße, AndyAm 05.04.2024 um 19:05 schrieb mleck28 ***@***.***>:﻿
Hallo kirchner85,
bist du noch an einer Fahrzeugerkennung interessiert?

**Comment by mleck28** [comment]
Bist Du mit Node-Red unterwegs?

**Comment by kirchner85** [comment]
Genau,primär nutzte ich Node-RED (Verbindung zur Renault-API selbst gebaut, primär MQTT für die unterschiedlichsten Status vom go-eCharger, Web-Calls für das Steuern ob geladen werden soll), habe aber auch noch einen HomeAssistant am laufen.Viele GrüßeAm 06.04.2024 um 19:13 schrieb mleck28 ***@***.***>:﻿
Bist Du mit Node-Red unterwegs?

**Reply by 4homeassistant** [reply]
Wir haben das Ganze mit Home Assistant als Automation je Fahrzeug gelöst. Es gibt bei uns 3 Fahrzeuge, 2 davon lassen sich über API pollen (Hyundai Ioniq 5/6, Audi A6 e-tron), 1 nicht (Cupra) wobei hier die API recht zügig liefert. Der Trigger bei den Fahrzeugen mit API Polling ist der jeweilige "car" State den go-e über API bzw. MQTT liefert, dann wird ein Refresh der Fahrzeug API ausgelöst, etwas gewartet und sofern der entsprechende Plug State des Fahrzeugs signalisiert, dass der Stecker verbunden ist wird "trx" gesetzt. Bei dem Fahrzeug ohne API-Polling läuft es umgedreht: der Trigger ist der Plug State des Fahrzeugs. Wenn der ausgelöst wurde, wird geprüft bei welchem der Charger in den letzten 5 Minuten "car" verbunden wurde. Funktioniert gut mit 2 Chargern und 3 Autos.
